src/dataframerow/dataframerow.jl

```diff
@@ -50,25 +50,26 @@ function Base.hash(r::DataFrameRow, h::UInt)
     return h
 end
 
-# compare two elements in the array
-_isequalelms(a::Array, i::Int, j::Int) = isequal(a[i], a[j])
-
-# compare the two elements in the data array
-function _isequalelms(a::DataArray, i::Int, j::Int)
-    if isna(a, i)
-        return isna(a, j)
-    else
-        return !isna(a, j) && isequal(a.data[i], a.data[j])
-    end
-end
-
-# compare two elements in the pooled array
-# NOTE assume there are no duplicated elements in the pool
-_isequalelms(a::PooledDataArray, i::Int, j::Int) = isequal(a.refs[i], a.refs[j])
-
 # comparison of DataFrame rows
 # only the rows of the same DataFrame could be compared
 # rows are equal if they have the same values (while the row indices could differ)
+function Base.:(==)(r1::DataFrameRow, r2::DataFrameRow)
+    if r1.df !== r2.df
+        throw(ArgumentError("Comparing rows from different frames not supported"))
+    end
+    if r1.row == r2.row
+        return Nullable(true)
```

Comment:
Why do we return `Nullable(true)` instead of just `true`? Is there something inherently `Nullable` about a DataFrameRow?